x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java

```diff
@@ -65,6 +65,8 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
          * isn't then we use the {@code Content-Type} header which is required.
          */
         String accept = null;
+        XContentType acceptHeader = null;
+        XContentType contentTypeHeader = null;
```

Comment:
Wondering if this media-type origin split is strictly necessary: the previous code tried to fetch a token from somewhere - in order of preference: enforced format (CBOR), `format` URL attribute, `Accept` header, `Content-Type` header and repeatedly checking `accept` against `null` - then validate that token against a _media type_ or as a _format_ (#`fromMediaTypeOrFormat()`).

The new code checks if it's a _format_ (in `getXContentType()` @ L143), or then if it's a _media type_, by one of these XContentType instances (@ L147). **But** the source preferences is already stored in the way the `accept` member is set (by those subsequent `null` checks), so could we not simply serially invoke #`fromMediaType(accept)` and #`fromFormat(accept)` and return the first non-null value to simplify the code? Similar to what we do in `TextFormat#fromMediaTypeOrFormat()`.

Reply:
good point, I have refactored this as per your suggestions